addons/mail/static/src/models/message_action_view.js

```diff
@@ -0,0 +1,216 @@
+                    } else {
+                        classNames.push('fa-star-o');
+                    }
+                    break;
+                default:
+                    break;
+            }
+            return classNames.join(' ');
+        },
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computePaddingClassNames() {
+            const isDeviceSmall = this.messaging.device.isSmall;
+            const paddingClassNames = [];
+
+            if (this.messageAction.messageActionListOwner.firstActionView === this) {
+                isDeviceSmall ? paddingClassNames.push('ps-3') : paddingClassNames.push('ps-2');
+            } else {
+                isDeviceSmall ? paddingClassNames.push('ps-2') : paddingClassNames.push('ps-1');
+            }
+
+            if (this.messageAction.messageActionListOwner.lastActionView === this) {
+                isDeviceSmall ? paddingClassNames.push('pe-3') : paddingClassNames.push('pe-2');
+            } else {
+                isDeviceSmall ? paddingClassNames.push('pe-2') : paddingClassNames.push('pe-1');
+            }
+
+            if (isDeviceSmall) {
+                paddingClassNames.push('py-3')
+            } else {
+                paddingClassNames.push('py-2')
+            }
+    
+            return paddingClassNames.join(' ');
+        },
+        /**
+         * @private
+         * @returns {integer|FieldCommand}
+         */
+        _computeTabindex() {
+            if (this.messageAction.messageActionListOwnerAsReaction) {
+                return clear();
+            }
+            return 0;
+        },
+        /**
+         * @private
+         * @returns {string|FieldCommand}
+         */
+        _computeTitle() {
+            switch (this.messageAction.messageActionListOwner) {
+                case this.messageAction.messageActionListOwnerAsDelete:
+                    return this.env._t("Delete");
+                case this.messageAction.messageActionListOwnerAsEdit:
+                    return this.env._t("Edit");
+                case this.messageAction.messageActionListOwnerAsMarkAsRead:
+                    return this.env._t("Mark as Read");
+                case this.messageAction.messageActionListOwnerAsReaction:
+                    return this.env._t("Add a Reaction");
+                case this.messageAction.messageActionListOwnerAsReplyTo:
+                    return this.env._t("Reply");
+                case this.messageAction.messageActionListOwnerAsToggleCompact:
+                    return this.env._t("Compact");
+                case this.messageAction.messageActionListOwnerAsToggleStar:
+                    return this.env._t("Mark as Todo");
+                default:
+                    return clear();
+            }
+        },
+    },
+    fields: {
+        /**
+         * States the reference to the action in the component. Used in `Reaction` action
```

Comment:
"used in" is typically not a good comment (might change and then it gets obsolete)